src/averageAges.js

```diff
@@ -55,7 +85,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenList = people.filter(child => {
+    return onlyWithSon
+      ? people.some(mother => mother.name === child.mother && child.sex === 'm')
+      : people.some(mother => mother.name === child.mother);
+  });
+
+  const ageDiffSum = childrenList.reduce((acc, child) => {
+    const foundMother = people.find(mother =>
+      mother.name === child.mother
+    );
+
+    if (foundMother) {
+      const ageDiff = child.born - foundMother.born;
+
+      return acc + ageDiff;
+    }
+
+    return acc;
+  }, 0);
+
+  return ageDiffSum / childrenList.length;
+}
+
+function getTotalAge(arrayOfPeople) {
+  return arrayOfPeople.reduce((sumOfAge, person) => {
+    const { born, died } = person;
+
+    return sumOfAge + (died - born);
+  }, 0);
 }
```

Comment:
Extend you function to calculate average ages, not only total age.

```suggestion
function getAverageAge(ages, divider) {
  return ages.reduce((sum, age) => (sum + age, 0) / divider || 0;
}
```

Reply:
@33Orange to be honest it's beautiful